src/assign.rs

```diff
@@ -380,7 +514,7 @@ mod json {
             ];
 
             for (path, val, expected, expected_replaced) in tests {
-                let ptr = PointerBuf::parse(path).expect(&format!("failed to parse \"{path}\""));
+                let ptr = PointerBuf::must_parse(path);
```

Comment:
Why a `PointerBuf` here?

Reply:
I don't know :)

I think the old tests used them because they were ported from the old code. I just kept chugging away with the way things were. Fixing it.